lib/smart_proxy_dns_powerdns/dns_powerdns_configuration_validator.rb

```diff
@@ -0,0 +1,37 @@
+require 'smart_proxy_dns_powerdns/dns_powerdns_plugin'
+
+module Proxy::Dns::Powerdns
+  class ConfigurationValidator
+    def validate_settings!
+      validate_choice(:powerdns_backend, ['mysql', 'postgresql', 'dummy'])
+
+      case settings[:powerdns_backend]
+      when 'mysql' then
+        validate_presence :powerdns_mysql_username, :powerdns_mysql_password, :powerdns_mysql_database
+      when 'postgresql' then
+        validate_presence :powerdns_postgresql_connection
```

Comment:
Because I was unsure about dealing with optional parameters. The connection string is easier because it can also handle socket connections.